frontend/server/k8s-helper.ts

```diff
@@ -184,11 +184,11 @@ export function waitForTensorboardInstance(logdir: string, timeout: number): Pro
   });
 }
 
-export function getPodLogs(podName: string): Promise<string> {
+export function getPodLogs(podName: string, podNamespace?: string): Promise<string> {
   if (!k8sV1Client) {
     throw new Error('Cannot access kubernetes API');
   }
-  return (k8sV1Client.readNamespacedPodLog(podName, namespace, 'main') as any).then(
+  return (k8sV1Client.readNamespacedPodLog(podName, podNamespace || namespace, 'main') as any).then(
```

Comment:
Do we need to update the global variable: namespace such that the getArgoWorkflow and getK8sSecret also work?

Reply:
We need to support those one by one. Global variable is shared and could be modified during processing of a request.

I don't currently know where those methods are used. I guess I should double check and make sure other features don't break.

Reply:
e.g. I searched that https://github.com/kubeflow/pipelines/blob/master/frontend/server/workflow-helper.ts#L147 is the usage of the workflow-helper calling the getArgoWorkflow. Could you proof-check if this functions are called in other places.

Reply:
Thanks, that was added in https://github.com/kubeflow/pipelines/pull/2081 to get pod log using argo workflow log persistence capability. I don't think we need to support this in multi user mode right now

Reply:
I verified these methods are only used for the use case I mentioned above. It's fine we defer these support.

Reply:
SG